go/extractor/cli/go-autobuilder/go-autobuilder.go

```diff
@@ -272,6 +272,11 @@ func main() {
 		} else if util.DirExists("vendor") {
 			modMode = ModMod
 		}
+		if util.FileExists("go.work") {
```

Comment:
Hey @cardil! 👋🏼 Thank you for your contribution. Checking for the presence of a `go.work` file should be sufficient in most cases, but I have a suggestion to make the check for workspace mode more resilient. 

Please consider checking the output of the `go env GOWORK` command instead. [This documentation](https://go.dev/blog/get-familiar-with-workspaces#:~:text=Workspace%20mode%20is%20enabled%20if%20the%20GOWORK%20variable%20names%20a%20path%20to%20a%20file%20that%20ends%20in%20.work.%20To%20determine%20which%20go.work%20file%20is%20being%20used%2C%20run%20go%20env%20GOWORK.%20The%20output%20is%20empty%20if%20the%20go%20command%20is%20not%20in%20workspace%20mode.) from the Go blog provides some context on the command.

Checking for workspace mode using `go env GOWORK` is resilient to the build being run with `GOWORK=off`, in which case workspace mode should _not_ be used even if a `go.work` file is present. Please let us know if you have any questions about this suggestion, and thank you again for opening this PR.

Reply:
If it helps, there are existing patterns within that file for executing a command and working with the output. Here's an example: https://github.com/github/codeql/blob/862948f1ccbd2b1aac9c678a705aeecfb271136a/go/extractor/cli/go-autobuilder/go-autobuilder.go#L317-L321

Reply:
Make sense.

Although I'm thinking about tests. I can't see any. Maybe I should include some?!

Reply:
A contributions that includes one or more tests is always a bonus. 😄 We have a couple of existing tests in `go-autobuilder_test.go`, for functions within the `go-autobuilder.go` file. If you do intend to include tests, that would be the place to do it.

Reply:
We'll probably want an end-to-end integration test for these, and we're just converting those tests for Go into a new format, so it's probably best to leave writing that test to us just now

Reply:
Those tests have been converted now, in this PR: https://github.com/github/codeql/pull/12130